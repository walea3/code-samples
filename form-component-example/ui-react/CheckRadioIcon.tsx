import React from 'react';
import { CheckRadioProps } from './';

interface CheckRadioIconProps extends React.SVGAttributes<SVGElement> {
  variant: CheckRadioProps['variant'];
}

export const IconPaths = {
  radio:
    'M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z',
  checkbox: 'M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z',
};

export function CheckRadioIcon({ variant, ...props }: CheckRadioIconProps) {
  const svgPath: string = IconPaths[variant];

  const accessibilityProps = {
    focusable: false,
    'aria-hidden': true,
  };

  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 28 28"
      {...props}
      {...accessibilityProps}
    >
      <path d={svgPath} />
    </svg>
  );
}
