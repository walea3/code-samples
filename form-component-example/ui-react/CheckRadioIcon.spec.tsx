import { renderWithNEL } from '@nel-ui/tooling-react/test-utils';
import React from 'react';

import { CheckRadioIcon, IconPaths } from './CheckRadioIcon';

describe('<CheckRadioIcon />', () => {
  const renderComponent = (props) => renderWithNEL(<CheckRadioIcon {...props} />);

  describe('@renders', () => {
    test.each`
      name             | props
      ${'as radio'}    | ${{ variant: 'radio' }}
      ${'as checkbox'} | ${{ variant: 'checkbox' }}
    `('variant $name', ({ props }) => {
      expect(renderComponent(props)).toMatchSnapshot();
    });

    test('available CheckIcon variants', () => {
      expect(IconPaths).toMatchSnapshot();
    });
  });
});
