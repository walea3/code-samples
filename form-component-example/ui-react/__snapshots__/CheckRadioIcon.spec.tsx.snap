// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckRadioIcon /> @renders available CheckIcon variants 1`] = `
Object {
  "checkbox": "M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z",
  "radio": "M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z",
}
`;

exports[`<CheckRadioIcon /> @renders variant as checkbox 1`] = `
<svg
  aria-hidden="true"
  focusable="false"
  viewBox="0 0 28 28"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
  />
</svg>
`;

exports[`<CheckRadioIcon /> @renders variant as radio 1`] = `
<svg
  aria-hidden="true"
  focusable="false"
  viewBox="0 0 28 28"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M14,19L14,19c-2.8,0-5-2.2-5-5v0c0-2.8,2.2-5,5-5h0c2.8,0,5,2.2,5,5v0C19,16.8,16.8,19,14,19z"
  />
</svg>
`;
